File: Flutter Apps/delivery_vpharma/lib/Model/Warehouse.dart
class Warehouse {
  dynamic WHName;
  dynamic Address;
  dynamic PhoneNum;
  Warehouse({this.Address, this.PhoneNum, this.WHName});

  factory Warehouse.fromJson(dynamic responsedata) {
    return Warehouse(
      Address: responsedata['Address'],
      PhoneNum: responsedata['PhoneNum'],
      WHName: responsedata['WHName'],
    );
  }
}

File: app/Models/Warehouse.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Warehouse extends Model
{
    protected $table = 'warehouses';
    protected $primaryKey = 'ID';
    protected $fillable = ['WHLicense', 'WHName', 'Address', 'PhoneNum', 'status', 'user_id'];

    use HasFactory;

    public function medicines()
    {
        return $this->belongsToMany(Medicine::class, 'medicine_warehouses', 'warehouseID', 'MedicineID')
            ->withPivot('Price', 'Quantity');
    }
    public function pharmacyOrders()
    {
        return $this->hasMany(PharmacyOrder::class, 'warehouseID');
    }
}
